core/runtime.rs

```diff
@@ -822,6 +822,12 @@ impl JsRuntime {
         );
         let promise = v8::Local::<v8::Promise>::try_from(value)
           .expect("Expected to get promise as module evaluation result");
+        let empty_fn = |_scope: &mut v8::HandleScope,
+                        _args: v8::FunctionCallbackArguments,
+                        _rv: v8::ReturnValue| {};
+        let empty_fn = v8::FunctionTemplate::new(scope, empty_fn);
+        let empty_fn = empty_fn.get_function(scope).unwrap();
+        promise.catch(scope, empty_fn);
         let promise_global = v8::Global::new(scope, promise);
         let mut state = state_rc.borrow_mut();
         state.pending_promise_exceptions.remove(&promise_global);
```

Comment:
Out of curiosity, does adding this handler still makes it so that promise ends up in `state.pending_promise_exceptions`? If not we could cut some complexity thanks to this fix

Reply:
It still gets put in there if there is an immediate rejection (the cases that worked correctly before this patch), the promise reject callback gets called before the handler is added. Actually... I think adding the handler here triggers https://github.com/denoland/deno/blob/555595e6d02d1fc1adb535df074d258389fab509/core/bindings.rs#L252-L254 so maybe the remove below is redundant. Will investigate.